packages/payload/src/admin/components/elements/WhereBuilder/Condition/Number/index.tsx

```diff
@@ -3,20 +3,78 @@ import { useTranslation } from 'react-i18next'
 
 import type { Props } from './types'
 
+import ReactSelect from '../../../ReactSelect'
 import './index.scss'
 
 const baseClass = 'condition-value-number'
 
-const NumberField: React.FC<Props> = ({ disabled, onChange, value }) => {
+const NumberField: React.FC<Props> = ({ disabled, onChange, operator, value }) => {
   const { t } = useTranslation('general')
-  return (
+
+  const isMulti = ['in', 'not_in'].includes(operator)
+
+  const [valueToRender, setValueToRender] = React.useState<
+    { id: string; label: string; value: { value: number } }[]
+  >([])
+
+  const onSelect = React.useCallback(
+    (selectedOption) => {
+      let newValue
+      if (!selectedOption) {
+        newValue = []
+      } else if (isMulti) {
+        if (Array.isArray(selectedOption)) {
+          newValue = selectedOption.map((option) => Number(option.value?.value || option.value))
+        } else {
+          newValue = [Number(selectedOption.value?.value || selectedOption.value)]
+        }
+      }
+
+      onChange(newValue)
+    },
+    [isMulti, onChange],
+  )
+
+  React.useEffect(() => {
+    if (Array.isArray(value)) {
+      setValueToRender(
+        value.map((val, index) => {
+          return {
+            id: `${val}${index}`, // append index to avoid duplicate keys but allow duplicate numbers
+            label: `${val}`,
+            value: {
+              toString: () => `${val}${index}`,
+              value: (val as any)?.value || val,
+            },
+          }
+        }),
+      )
+    } else {
+      setValueToRender([])
+    }
+  }, [value])
+
+  return isMulti ? (
+    <ReactSelect
+      disabled={disabled}
+      isClearable
+      isCreatable
+      isMulti={isMulti}
+      isSortable
+      numberOnly
+      onChange={onSelect}
+      options={[]}
+      placeholder={t('enterAValue')}
```

Comment:
lets use general:enterAValue for easier portability into 3.0.

I would do that everywhere in this file. Instead of scoping it with const { t } = useTranslation('general')